From 6f50dd1f7ed95a33298192023ccae8f61b648972 Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@nxp.com>
Date: Wed, 12 Sep 2018 21:26:39 -0500
Subject: [PATCH 4622/5242] MLK-19560: drm: imx: hdp: Display Port PHY
 reference needs to be enabled

commit  8f7d68c8756bdd0da4455dac27d83d3c0561d9ca from
https://source.codeaurora.org/external/imx/linux-imx.git

The pixel_clock_set_rate function must be called before enabling the pixel clock.
This will enable the PHY reference clock for DisplayPort mode.

Signed-off-by: Oliver Brown <oliver.brown@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index bf48eb8..7f22ee1 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -1384,6 +1384,9 @@ static int imx_hdp_imx_bind(struct device *dev, struct device *master,
 		DRM_ERROR("Failed to initialize IPG clock\n");
 		return ret;
 	}
+
+	imx_hdp_call(hdp, pixel_clock_set_rate, &hdp->clks);
+
 	imx_hdp_call(hdp, pixel_clock_enable, &hdp->clks);
 
 	imx_hdp_call(hdp, phy_reset, hdp->ipcHndl, &hdp->mem, 0);
-- 
1.7.9.5

